torch/_dynamo/variables/functions.py

```diff
@@ -573,6 +575,78 @@ def call_method(
                 # https://github.com/python/cpython/pull/104771
                 assert tracer.symbolic_result is not None
                 return tracer.symbolic_result
+        elif name == "throw":
+            # * Raises an exception at the point where the generator was paused, and
+            # returns the next value yielded by the generator.
+            # * If the generator exits without yielding, raise StopIteration
+            # * If the generator function does not catch the passed-in exception,
+            # or raises a different exception, then that exception propagates to the caller.
+
+            if len(args) > 1:
+                raise IncorrectUsage(
+                    "the (type, exc, tb) signature of throw() is deprecated, "
+                    "use the single-arg signature instead."
+                )
+
+            # Setup the exception table and jump target in case of try...finally
+            tracer = self._get_inline_tracer(tx)
+            try:
+                self._setup_exception(tx, args[0])
+            except ObservedException:
+                # propagate the exception back to the parent caller
+                tx.exn_vt_stack.extend(tracer.exn_vt_stack)
+                raise
+
+            retval = self.next_variable(tx)
+
+            # We need to re-raise the exception one more time. But why? Let’s walk
```

Comment:
Btw if we go with always raising an exception I think I might prefer doing a generator.close() or raising a GeneratorExit here

Reply:
nvm, your explanation today made sense to me